### DISCUSSION TITLE
Have option to define RAPIDJSON_DEFAULT_ALLOCATOR

### DISCUSSION THREAD

**Discussion by Habbus** [c-jimenez/open-ocpp#165] [General]
Hi,

With the latest versions of rapidjson that I find on github, there is an option to manually define the memory allocator with a macro (this is from [github](https://github.com/Tencent/rapidjson/blob/f9d53419e912910fd8fa57d5705fa41425428c35/include/rapidjson/document.h#L71-L79)):
```cpp
// ... rapidjson/include/rapidjson/document.h

/*! \def RAPIDJSON_DEFAULT_ALLOCATOR
    \ingroup RAPIDJSON_CONFIG
    \brief Allows to choose default allocator.

    User can define this to use CrtAllocator or MemoryPoolAllocator.
*/
#ifndef RAPIDJSON_DEFAULT_ALLOCATOR
#define RAPIDJSON_DEFAULT_ALLOCATOR ::RAPIDJSON_NAMESPACE::MemoryPoolAllocator<::RAPIDJSON_NAMESPACE::CrtAllocator>
#endif
```

In the current version of rapidjson placed in the 3rdparty directory, this option does not seem to be there.
Would it be possible to have this as an option (maybe in the [CMakeLists_Options.txt](https://github.com/c-jimenez/open-ocpp/blob/develop/CMakeLists_Options.txt))? I have been experiencing some problems with the MemoryPoolAllocator, while the CrtAllocator works perfectly.

I tested this by manually changing the templated Allocators over [here](https://github.com/c-jimenez/open-ocpp/blob/33667077162f24ee8f7162184aad16c906152a0e/3rdparty/rapidjson/include/rapidjson/document.h#L539) and [here](https://github.com/c-jimenez/open-ocpp/blob/33667077162f24ee8f7162184aad16c906152a0e/3rdparty/rapidjson/include/rapidjson/document.h#L2023).
Also, it requires an update to the [IMewssageConverter](https://github.com/c-jimenez/open-ocpp/blob/33667077162f24ee8f7162184aad16c906152a0e/src/messages/IMessageConverter.h#L269-L272).

**Comment by Habbus** [comment]
@c-jimenez what do you think about this?

**Reply by c-jimenez** [reply]
Hi,

Sorry, I missed your question.

The rapidjson library in the 3rdparty folder is the latest available tag => quite old but stable version.
You can try to switch to a newer version and add the compile option : if it doesn't trigger  regressions we can merge it in the develop branch.

Regards

**Reply by Habbus** [reply]
Thanks, I'll have a look at some point

**Comment by Habbus** [comment]
See #182
